source/server/converters.h

```diff
@@ -317,6 +318,19 @@ typename TypeToStorageType<TDriverType, TGrpcType>::StorageType allocate_output_
 }
 
 }  // namespace converters
+
+const int kMaxGrpcErrorDescriptionSize = 4096;
+
+inline ::grpc::Status ApiErrorAndDescriptionToStatus(google::protobuf::int32 status, std::string& description)
+{
+    converters::trim_trailing_nulls(description);
+    // TODO: May need full-fledged json escape logic.
+    for (size_t index = 0; (index = description.find('"', index)) != std::string::npos; ++index) {
+        description.replace(index, 1, "\\\"");
```

Comment:
I'm pretty sure you'll have an infinite loop here (until you run out of memory). If I'm reading it right, when you find a `"` character, you replace it with `\"` and then you still only increment `index` by 1, *which means you'll immediately find the `"` character you just [re]added and escape it again*!

(Also, instead of using `replace(index, 1, "\\\"")` you could use `insert(index, '\\')`, if you want.)

Reply:
If you don't want to add a dependency on a full-fledged JSON parser, consider the `encodeAndStrcpy` function from jsonz.

Reply:
That sounds good. I'll add to the comment. It won't happen in this PR because I have a separate task for it and I'll want to take time to add good test cases for it.

Reply:
David I will increment by 2. While your suggestion to use `insert` is tantalizing, that takes an iterator instead of an index. :(

Reply:
`grpc-device` already uses `nlohmann::json` to parse the config file.

Reply:
"I will increment by 2" -- careful there. You shouldn't have a "+= 2" anywhere here. You just need an extra ++index after doing the `replace`, and when that's combined with the for loop's ++index it'll add up to 2.

Reply:
Meaning, I'll change `++index` in the for loop to `index += 2` because it only ever loops around when it does a replace.

Reply:
> `grpc-device` already uses `nlohmann::json` to parse the config file.

Great, using that library to produce this JSON string is literally a one-liner.

https://godbolt.org/z/vT1hajGnj
```
#include <iostream>
#include <nlohmann/json.hpp>

std::string createStatusJson(int code, std::string message)
{
    return to_string(nlohmann::json{{"code", code}, {"message", message}});
}
```